protostar/cheatable_starknet/cheatable_entry_point.py

```diff
@@ -258,6 +261,23 @@ def execute(
         resources_manager: ExecutionResourcesManager,
         general_config: StarknetGeneralConfig,
         tx_execution_context: TransactionExecutionContext,
+    ):
+        try:
+            return self._execute(
+                state,
+                resources_manager,
+                general_config,
+                tx_execution_context,
+            )
+        except StarkException as ex:
+            raise self._map_stark_exception(ex)
+
+    def _execute(
```

Comment:
do we need to create an extra function for this? Wouldn't it be enough to wrap `return super().execute(...` with a `try ... except`?

Reply:
> do we need to create an extra function for this?

It's not needed. I created it, because it's a good practice.

> Try block should have one line.
https://gist.github.com/bansalankit92/6119c4f7fabe11b7a174a99f5a7bb376#exceptions-error-handling

> I don't see why the logic with max_step cannot be in the execute method (instead of _execute) and why it should be in the try ... except block, it doesn't seem to raise a StarkException.
> Wouldn't it be enough to wrap return super().execute(... with a try ... except?

It would be better. The existing code in the `execute` method wasn't super obvious, I used [wrap method technique](https://gist.github.com/jonnyjava/42883d4e464167f81e2ee60a488a5ded#wrap-method) to avoid touching it. However, after looking at the code in the `_execute` method again, it looks like the `max_steps` code doesn't have an impact anyway. 

[removed _execute, refined code in the CheatableEntryPoint](https://github.com/software-mansion/protostar/pull/1404/commits/e99e4f26a08d14c7b8a95fdbe0fa6cf116254be0)

Reply:
[make max_steps affect execute](https://github.com/software-mansion/protostar/pull/1404/commits/d67e4d972358bcf4cda51c0ad6e1e73690d7ede5)

Reply:
Makes sense, looks better now